pkg/remote/kyma_synchronization_context.go

```diff
@@ -186,24 +187,47 @@ func (c *KymaSynchronizationContext) CreateOrFetchRemoteKyma(
 		if !kyma.DeletionTimestamp.IsZero() {
 			return nil, ErrNotFoundAndKCPKymaUnderDeleting
 		}
-		kyma.Spec.DeepCopyInto(&remoteKyma.Spec)
 
-		// if KCP Kyma contains some modules during initialization, not sync them into remote.
-		remoteKyma.Spec.Modules = []v1beta2.Module{}
+		if err = c.copyOldKymaCRIfExisting(ctx, kyma, remoteSyncNamespace, remoteKyma, recorder); err != nil {
+			kyma.Spec.DeepCopyInto(&remoteKyma.Spec)
+			// if KCP Kyma contains some modules during initialization, not sync them into remote.
+			remoteKyma.Spec.Modules = []v1beta2.Module{}
+		}
 
 		err = c.RuntimeClient.Create(ctx, remoteKyma)
 		if err != nil {
 			recorder.Event(kyma, "Normal", "RemoteInstallation", "Kyma was installed to SKR")
 
 			return nil, err
 		}
+
 	} else if err != nil {
 		recorder.Event(kyma, "Warning", err.Error(), "Client could not fetch remote Kyma")
 
 		return nil, err
 	}
 
-	return remoteKyma, err
+	return remoteKyma, nil
+}
+
+func (c *KymaSynchronizationContext) copyOldKymaCRIfExisting(
+	ctx context.Context, kyma *v1beta2.Kyma, remoteSyncNamespace string,
+	remoteKyma *v1beta2.Kyma, recorder record.EventRecorder) error {
+	oldRemoteKyma := &v1beta2.Kyma{}
+	oldRemoteKyma.Name = kyma.Name
+	oldRemoteKyma.Namespace = remoteSyncNamespace
+	err := c.RuntimeClient.Get(ctx, client.ObjectKeyFromObject(oldRemoteKyma), oldRemoteKyma)
```

Comment:
here we need to consider the error differently, if the error is not found, we can return, but if it's other error, it might be temporary can't reach the oldRemoteKyma, but it still exists, which means we should not create new kyma, because after the new kyma get created, there is no chance to delete this old kyma anymore.